### PR TITLE
Bugfix ingest feb

### PR DESCRIPTION
**Pull request by xiphl**
reference #55 
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: ingest-api/ingest_api/helper/security.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    os.environ["DATAHUB_FRONTEND"] = "http://172.19.0.1:9002"
+
+
+def verify_token(token: str, user: str):
+    token_secret = os.environ["JWT_SECRET"]
+    # log.debug(f"signature secret is {token_secret}")
+    try:
+        payload = jwt.decode(token, token_secret, algorithms="HS256")
+        # userUrn = impossible is set in the frontend-react as a temporary userUrn while graphQL is returning identity info.
+        if payload["actorId"] == "impossible":
```
Comment (neojunjie) [inline ingest-api/ingest_api/helper/security.py:31]: When will this check be triggered?
If not it is redundant to check for it?

File: ingest-api/ingest_api/helper/mce_convenience.py [hunk 2]
```diff
@@ -133,48 +127,6 @@ def derive_platform_name(input: str) -> str:
     platform_name = f"urn:li:dataPlatform:{platform}"
     return platform_name
 
-
```
Comment (neojunjie) [inline ingest-api/ingest_api/helper/mce_convenience.py:136]: this code block no longer valid ar?

File: ingest-api/ingest_api/config.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+from prometheus_client import multiprocess
+
+def child_exit(server, worker):
```
Comment (neojunjie) [inline ingest-api/ingest_api/config.py:3]: we need to force kill the process?

File: ingest-api/ingest_api/helper/security.py (hunk 1 above)
Comment (xiphl) [reply ingest-api/ingest_api/helper/security.py:31]: Frontend there is a set of whoami functions, where the default username is impossible when graphql hasn't finished resolving the user identity. If the user submit damn fast, possibly will be triggered to submit this? It should be impossible to reach this state though.

File: ingest-api/ingest_api/config.py (hunk 3 above)
Comment (xiphl) [reply ingest-api/ingest_api/config.py:3]: Is a gunicorn server config required, the guide say need to add so I added. 
Gunicorn is used as process manager to spin up multiple processes to service requests.

File: docker/ingest-api/env/docker.env [hunk 4]
```diff
@@ -6,4 +6,8 @@ DATAHUB_AUTHENTICATE_INGEST=yes
 # use either 'True' or 'False' cos environ var has no bool.
 # frontend is for graphql queries for ownership
 DATAHUB_FRONTEND=http://172.19.0.1:9002
-ELASTIC_HOST=http://elasticsearch:9200
\ No newline at end of file
+ELASTIC_HOST=http://elasticsearch:9200
+DATAHUB_BACKEND=http://172.19.0.1:8080
+ANNOUNCEMENT_URL=https://xaluil.gitlab.io/announce/
+DATASET_PROFILE_INDEX=dataset_datasetprofileaspect_v1
+PROMETHEUS_MULTIPROC_DIR=/gunicorn/
```
Comment (neojunjie) [inline docker/ingest-api/env/docker.env:13]: mandatory parameters?

File: datahub-web-react/src/app/entity/dataset/whoAmI.tsx [hunk 5]
```diff
@@ -14,14 +14,49 @@ export function FindMyUrn() {
     return data?.me?.corpUser.urn ?? '';
 }
 
+export function FindMyGroups() {
+    const input = FindMyUrn();
+    console.log(`myurn is ${input}`);
+    const queryresult = gql`
+        query test($urn: String!) {
```
Comment (neojunjie) [inline datahub-web-react/src/app/entity/dataset/whoAmI.tsx:21]: better name for the query method

File: ingest-api/ingest_api/helper/mce_convenience.py (hunk 2 above)
Comment (xiphl) [reply ingest-api/ingest_api/helper/mce_convenience.py:136]: I've not added platform instances or the new data classes yet, still riding on the old MCE format

File: datahub-web-react/src/app/announce/BannerSplash.tsx [hunk 6]
```diff
@@ -4,21 +4,23 @@ import { Agent } from 'https';
 import { Alert } from 'antd';
 import axios from 'axios';
 import ReactHtmlParser from 'react-html-parser';
-import announceConfig from '../../conf/Announcement';
```
Comment (neojunjie) [inline datahub-web-react/src/app/announce/BannerSplash.tsx:7]: can make it such that I can enable/disable announcement feature?

**Review by neojunjie — COMMENTED** [review]
Yoz.... added some comments in the files

**Review by neojunjie — APPROVED** [review]

**Pull request merged**